include/nbl/system/IFileArchive.h

```diff
@@ -88,10 +88,10 @@ class IFileArchive : public core::IReferenceCounted
 			return &(*found);
 		}
 
-		std::mutex itemMutex; // TODO: update to readers writer lock
+		mutable std::mutex itemMutex; // TODO: update to readers writer lock
 		path m_defaultAbsolutePath;
 		// files and directories
-		core::vector<SListEntry> m_items;
+		mutable core::vector<SListEntry> m_items;
```

Comment:
why are these mutable now?

Reply:
ok I see, push these memebrs into `CFileArchive` (and make `getItemFromPath` use the `listAssets` or make it virtual)